packages/shared/pkg/logger/exporter.go

```diff
@@ -0,0 +1,113 @@
+package logger
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+type HTTPWriter struct {
+	ctx        context.Context
+	url        string
+	httpClient *http.Client
+	wg         sync.WaitGroup
+}
+
+func NewHTTPWriter(ctx context.Context, endpoint string) zapcore.WriteSyncer {
+	return &HTTPWriter{
+		ctx: ctx,
+		url: endpoint,
+		httpClient: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+		wg: sync.WaitGroup{},
+	}
+}
+
+func NewBufferedHTTPWriter(ctx context.Context, endpoint string) zapcore.WriteSyncer {
+	httpWriter := &zapcore.BufferedWriteSyncer{
+		WS:            NewHTTPWriter(ctx, endpoint),
+		Size:          256 * 1024, // 256 kB
+		FlushInterval: 5 * time.Second,
+	}
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			if err := httpWriter.Stop(); err != nil {
+				fmt.Printf("Error stopping HTTP writer: %v\n", err)
+			}
+		}
+	}()
+
+	return httpWriter
+}
+
+func (h *HTTPWriter) Write(source []byte) (n int, err error) {
+	h.wg.Add(1)
+
+	p := make([]byte, len(source))
```

Comment:
```suggestion
        // zap is reusing the buffer, so since we're processing it in a Go routine, we need to make a copy.
	p := make([]byte, len(source))
```

Reply:
Any reason why to not add it?